### PR TITLE
User interface for tic-tac-toe.

### PR DESCRIPTION
**Pull request by mahbubul14**
# tic_tac_toe in ruby program (milestone- 2)

## In this milestone we:
- Created a user_interface branch for milestone-2 
- Added content for the user interface
- Added validations to accept string for players username only
- Added validations to pick only 1-9  on board
- Add a loop in the game

### CODE REVIEW

**Review by mke2111 — CHANGES_REQUESTED** [review]
# STATUS: CHANGES REQUIRED 🔴 ♻️ 

## Hey Happy Coders, 👋🏽 
**This is a great project you have going on, well-done lads**. 

### Well-done 💯 
    1. The game requirements are met.
    2. The linter checks are a success.

**However, there are a few sections to improve before your project is approved.** 🔴 
<hr>

### REQUIRED

- [x] Add a more brief description message to the PR message body.
- [x] The game should inform the player if the selected move is invalid (without computing what an invalid move is). [screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/OOP/milestone2/invalid_move.png)
You need to validate the user input for the board positions(1-9), everything else should be considered invalid.
- [x] A loop is needed in this game. Inside the loop, you should call actions repeated for each move. The loop needs to be terminated at some point (after a winning move or draw to be more specific). It can be done with variable.
```
game_on = true

# initial instructions for player

while game_on
  # loop for each move

  if winner #  (the exact condition in this milestone is not important)
    game_on = false
  end

  if draw # (the exact condition in this milestone is not important)
    game_on = false
  end
end
```

#

### Kindly make all the requested changes and submit another CRR. 💯 

**Cheers 🍹 and Happy coding 💻** 

<hr>

Mention @mke2111 in this thread and I'll be glad to help. 😄 
### Best regards
Slack 🆘  Roy Mukuye
Twitter 🐦  [Roy Mukuye](https://twitter.com/Roymkenya)
LinkedIn 🔗  [Roy Mukuye](https://www.linkedin.com/in/roy-mukuye/)

File: bin/main
Comment (mke2111) [inline bin/main]: Add restrictions not to allow plain digits as usernames.

File: bin/main
Comment (mke2111) [inline bin/main]: Add restrictions to only allow single digits 1-9 since they are the only board positions

**Review by BrianSammit — CHANGES_REQUESTED** [review]
CHANGES REQUIRED  :red_circle: 
-------------
Hello @mahbubul14 @prolajumokeoni  you doing an amazing job in this milestone :muscle: :clap: 
However, there is some fix to make to complete this milestone :hammer_and_wrench: see it  below :eyes: :point_down: 
--------------

- [x] When I try to make a second move I got this error. 

Screenshot from 2021-05-12 17-19-17
--------------
 Please make the required change and send it again for another review :+1: 
 HAPPY CODING!!!

tenor (1)

**Review by ElijahTobs — CHANGES_REQUESTED** [review]
# Status: Changes Required :recycle: 

## Hi @mahbubul14 and @prolajumokeoni :wave: 

You've done well so far :+1:  and I commend you for that :clap: 
However, there is just a little change to be made before this milestone can be approved.

## HIGHLIGHTS 💯 
- Your PR message is well-detailed ✔️ 
- Descriptive commit messages ✔️ 

## REQUIRED

* [x] Your game should inform players if the selected move is a draw move (without computing draw). [Example screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/OOP/milestone2/game_tie.png)

## Happy Coding :computer: 

_Feel free to reach out to me on slack_ **`@Elijah Ayandokun`**
  **Elijah O. Ayandokun**
[Github](https://github.com/ElijahTobs) | [LinkedIn](https://www.linkedin.com/in/ayandokunelijah/) | [Twitter](https://twitter.com/ElijahTobs)

**Review by Berabjesus — APPROVED** [review]
# **Status: PROJECT APPROVED** 
Great job on implementing the changes required @prolajumokeoni and @mahbubul14 Keep it up.  :champagne: :champagne: 
**All the required tasks have been fulfilled in this project. I will go ahead and mark this as complete. 👍** 

I wish you all the very best in your next tasks:1st_place_medal:
#### **If you have any questions, let me know by leaving a comment here and mentioning me. I will be happy to help** 
**HAPPY CODING :computer:[Bereket Beshane](https://bereketababu.me)** | [LinkedIn](https://linkedin.com/in/bereketbeshane) | [GitHub](https://github.com/Berabjesus)

**Pull request merged**
